agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4496,19 +4508,24 @@ bool slave_thread::channel_selection_current_channel_restricted()
                       << "channel switch required";
             return true;
         }
-        auto is_restricted =
-            std::find_if(preference.channels.begin(), preference.channels.end(),
-                         [&channel](const beerocks::message::sWifiChannel &wifi_channel) {
-                             return channel == wifi_channel.channel;
-                         });
+        auto is_restricted = std::find_if(
+            preference.channels.begin(), preference.channels.end(),
+            [&](const beerocks::message::sWifiChannel &wifi_channel) {
```

Comment:
This lambda looks a lot like the other one. Are-they the same? If yes I would take the opportunity to make it a function